quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/config/SolverBuildTimeConfig.java

```diff
@@ -18,37 +19,53 @@
  * See also {@link SolverRuntimeConfig}
  */
 @ConfigGroup
-public class SolverBuildTimeConfig {
+public interface SolverBuildTimeConfig {
+
+    /**
+     * A classpath resource to read the specific solver configuration XML.
+     *
+     * If this property isn't specified, that solverConfig.xml is optional.
+     */
+    Optional<String> solverConfigXml();
 
     /**
      * Enable runtime assertions to detect common bugs in your implementation during development.
      * Defaults to {@link EnvironmentMode#REPRODUCIBLE}.
      */
-    @ConfigItem
-    public Optional<EnvironmentMode> environmentMode;
+    Optional<EnvironmentMode> environmentMode();
 
     /**
      * Enable daemon mode. In daemon mode, non-early termination pauses the solver instead of stopping it,
      * until the next problem fact change arrives. This is often useful for real-time planning.
      * Defaults to "false".
      */
-    @ConfigItem
-    public Optional<Boolean> daemon;
+    Optional<Boolean> daemon();
 
     /**
      * Determines how to access the fields and methods of domain classes.
      * Defaults to {@link DomainAccessType#GIZMO}.
      */
-    @ConfigItem
-    public Optional<DomainAccessType> domainAccessType;
+    Optional<DomainAccessType> domainAccessType();
+
+    /**
+     * Enable multithreaded solving for a single problem, which increases CPU consumption.
+     * Defaults to {@value SolverConfig#MOVE_THREAD_COUNT_NONE}.
+     * Other options include {@value SolverConfig#MOVE_THREAD_COUNT_AUTO}, a number
+     * or formula based on the available processor count.
+     */
+    Optional<String> moveThreadCount();
+
+    /**
+     * Configuration properties regarding Timefold's {@link TerminationConfig}.
+     */
+    TerminationRuntimeConfig termination();
 
     /**
      * What constraint stream implementation to use. Defaults to {@link ConstraintStreamImplType#BAVET}.
      *
      * @deprecated Not used anymore.
      */
-    @ConfigItem
     @Deprecated(forRemoval = true, since = "1.4.0")
-    public Optional<ConstraintStreamImplType> constraintStreamImplType;
+    Optional<ConstraintStreamImplType> constraintStreamImplType();
```

Comment:
Any chance to not include the deprecated field for the new config, only keep it for the old way? (If it's too hard to split the two, or requires code duplication, forget about it.)

Reply:
Can you please provide more details regarding this comment?

Reply:
This option is deprecated, which means we don't want new code and new users to be seeing it/using it.

For the "old" default solver config, we need to keep this option, because removing it would break backwards compatibility.
But technically, for the `@Named` configs (which are a new feature), we have no backwards compatibility to worry about, and therefore we can possibly expose an interface to those which does not include this deprecated option.

That's in theory. In practice, I worry this would've caused a lot of code duplication or otherwise some kind of a hack, so if that's the case, let's keep it as is and forget about this comment.